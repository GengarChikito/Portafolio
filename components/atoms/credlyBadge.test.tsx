import { describe, it, expect, vi } from "vitest";
import { render, screen, within } from "@testing-library/react";
import React from "react";
import CredlyBadge from './credlyBadge';


vi.mock("antd", () => {
    return {
        Card: ({ children, hoverable, ...rest }: any) => (
            <div data-testid="antd-card" {...rest}>{children}</div>
        ),
        Tooltip: ({ children }: any) => <>{children}</>,
    };
});



const props = {
    name: "Data Visualization Expert",
    issuer: "CertiProf",
    issued: "Mar 2024",
    imageUrl: "https://images.credly.com/badge.png",
    credentialUrl: "https://www.credly.com/badges/xyz",
};

describe("CredlyBadge", () => {
    it("renderiza el enlace envolviendo todo con href, target y rel correctos", () => {
        render(<CredlyBadge {...props} />);

        const link = screen.getByRole("link");
        expect(link).toBeInTheDocument();
        expect(link).toHaveAttribute("href", props.credentialUrl);
        expect(link).toHaveAttribute("target", "_blank");
        // sólo comprueba que incluya noreferrer (puede tener más valores)
        expect(link.getAttribute("rel")).toContain("noreferrer");

        // El contenido está dentro del link
        const card = within(link).getByTestId("antd-card");
        expect(card).toBeInTheDocument();
    });

    it("muestra la imagen con src y alt accesibles", () => {
        render(<CredlyBadge {...props} />);

        const img = screen.getByRole("img", { name: props.name });
        expect(img).toBeInTheDocument();
        expect(img).toHaveAttribute("src", props.imageUrl);
        expect(img).toHaveAttribute("alt", props.name);
    });

    it("muestra el nombre como heading y los metadatos de issuer/issued", () => {
        render(<CredlyBadge {...props} />);

        // h4 => role=heading level=4
        const heading = screen.getByRole("heading", { level: 4, name: props.name });
        expect(heading).toBeInTheDocument();

        // Texto combinado: "issuer · issued"
        expect(
            screen.getByText(`${props.issuer} · ${props.issued}`)
        ).toBeInTheDocument();
    });

    it("snapshot básico (estructura estable del output)", () => {
        const { container } = render(<CredlyBadge {...props} />);
        expect(container).toMatchSnapshot();
    });
});
