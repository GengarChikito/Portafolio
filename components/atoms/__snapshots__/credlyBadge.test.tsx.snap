// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CredlyBadge > snapshot básico (estructura estable del output) 1`] = `
<div>
  <a
    href="https://www.credly.com/badges/xyz"
    rel="noreferrer"
    style="text-decoration: none;"
    target="_blank"
  >
    <div
      data-testid="antd-card"
      style="border-radius: 16px; text-align: center;"
    >
      <img
        alt="Data Visualization Expert"
        src="https://images.credly.com/badge.png"
        style="width: 84px; height: 84px; object-fit: contain; margin: 8px auto;"
      />
      <h4
        style="margin: 8px 0px 4px; font-size: 14px; line-height: 1.2;"
      >
        Data Visualization Expert
      </h4>
      <small
        style="display: block; color: rgb(136, 136, 136);"
      >
        CertiProf
         · 
        Mar 2024
      </small>
    </div>
  </a>
</div>
`;
